ui/helpers/utils/permission.js

```diff
@@ -50,13 +54,28 @@ function getSnapNameComponent(snapName) {
   );
 }
 
-export const PERMISSION_DESCRIPTIONS = deepFreeze({
+const PERMISSION_DESCRIPTIONS = deepFreeze({
   // "endowment:caip25" entry is needed for the Snaps Permissions Review UI
-  [Caip25EndowmentPermissionName]: ({ t }) => ({
-    label: t('permission_ethereumAccounts'),
-    leftIcon: IconName.Eye,
-    weight: PermissionWeight.eth_accounts,
-  }),
+  [Caip25EndowmentPermissionName]: ({ t, permissionValue }) => {
+    const caveatValue = getRequestedSessionScopes({
+      permissions: permissionValue,
+    });
+    const requestedAccounts = getEthAccounts(caveatValue);
+    const isLegacySwitchChain = Boolean(!requestedAccounts.length);
```

Comment:
```suggestion
    const isLegacySwitchChain = requestedAccounts.length === 0;
```
Nit, but seems simpler to read?

Reply:
resolved in https://github.com/MetaMask/metamask-extension/pull/29824/commits/1ed12182ad388389de24cadca30f8fa35d16a683